lambdas/data-migration2/main.tf

```diff
@@ -0,0 +1,106 @@
+locals {
+  lambda_path = "${path.module}/dist/webpack/lambda.zip"
+}
+
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
```

Comment:
This concern sits somewhat squarely in between 'ease of use/what does this look as a user package' and 'nit': 

These roles duplicate a lot of the resources required for the migration, which only gets us a seperation-of-access benefit from having mig1 and mig2 have access to different tables.  

I'm wondering if that actually buys us anything over something like using the lambda processing mega-role that's being exported from Core, or at least making these two lambdas a combined tf-module.

Reply:
I think the only reason for separating them was that migration1 can be run easily as just a Lambda given the expected data volume/Lambda runtime, whereas migration2 will have to be an ECS task given the expected time necessary to migrate all granules/files. But both of those migrations could be deployed from a general `data-migration` parent module, I think. 

I'd prefer that and having them share a Lambda role that is defined once than use a "mega role" from core. I think it's just better to only allocate exactly the permissions you need for what you're doing

Reply:
Generically, I tend to agree, and think at some point we need to consider a more granular permission structure for the proc lambdas, despite the problems that presents.    As for if that drives user value....

Using a single module certainly reduces "AWS resource litter".